app/views/FileManagementView.scala.html

```diff
@@ -0,0 +1,59 @@
+@*
+ * Copyright 2024 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *@
+
+@import models.filemanagement.FileManagementTable
+@import viewmodels.FileManagementViewModel
+@import uk.gov.hmrc.govukfrontend.views.html.components.GovukTable
+@import uk.gov.hmrc.govukfrontend.views.Aliases.Table
+
+@this(
+        layout: templates.Layout,
+        govukTable: GovukTable
+)
+
+@(viewModel: FileManagementViewModel)(implicit request: Request[_], messages: Messages)
+
+@layout(pageTitle = titleNoForm(viewModel.title)) {
+
+    <h1 class="govuk-heading-l">@viewModel.heading</h1>
+
+    <p class="govuk-body">@viewModel.paragraph1</p>
+
+    @if(viewModel.isFiles) {
+
+        @viewModel.availableFilesTable.map { table: FileManagementTable =>
```

Comment:
i would have a check that there isn't a govuktable component we might be able to use instead

Reply:
Hi, I am using the govuktable component, the map you see there is just to extract the data out of the viewmodel to then feed it into the govuktable component

```
@viewModel.pendingFilesTable.map { table: FileManagementTable =>
            @govukTable(Table(
                caption = Some(table.caption),
                head = Some(table.headRows),
                rows = table.rows
            )
          )
        }
        
```
```
case class FileManagementViewModel(
  availableFilesTable: Option[AvailableFilesTable],
  pendingFilesTable: Option[PendingFilesTable]
)
```
```
case class PendingFilesTable()(implicit messages: Messages) extends FileManagementTable {
  // TODO: Add apply in companion object to create table rows from data
  override val caption: String          = messages("fileManagement.pendingFiles.table.caption")
  override val body: Option[String]     = Some(messages("fileManagement.pendingFiles.table.body"))
  override val headRows: Seq[HeadCell]  =
    Seq(
      messages("fileManagement.pendingFiles.table.header1"),
      messages("fileManagement.pendingFiles.table.header2")
    ).map { content =>
      HeadCell(content = Text(content))
    }
  override val rows: Seq[Seq[TableRow]] = Seq(
    Seq(
      TableRow(
        content = Text("time answer")
      ),
      TableRow(
        content = Text("file answer")
      )
    ),
    Seq(
      TableRow(
        content = Text("time answer")
      ),
      TableRow(
        content = Text("file answer")
      )
    )
  )
}
```